### PR TITLE
feat: replace SEO with metadata property

### PR DESCRIPTION
**Pull request by Xillians**
This is according to issue https://github.com/bjerkio/website/issues/623 . See task for detailed description to get the why of this PR. 

This part is implementing the metadata property that will be used by sanity 

Meaningful changes:
* ~~SEO object is removed~~ keeping SEO object for migration
* Metadata object is added to case
* Search Engine object is part of Metadata
* Social media object is added to Metadata

### CODE REVIEW

**Comment by simenandre** [comment]
Let's keep the SEO for now, so we have a safe migration

File: schemas/objects/meta-data.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  type: 'object',
+  fields: [
+    {
+        name: 'SearchEngine',
+        type: 'searchEngine',
+        validation: (Rule: { required: () => any; }) => Rule.required(),
+    },
+    {
+        name: 'Socialmedia',
+        type: 'socialMedia',
```
Comment (simenandre) [inline schemas/objects/meta-data.ts:15]:
```suggestion
  name: 'metadata',
  title: 'Metadata',
  type: 'object',
  fields: [
    {
        name: 'searchEngine',
        type: 'searchEngine',
        validation: (Rule: { required: () => any; }) => Rule.required(),
    },
    {
        name: 'socialMedia',
        type: 'socialMedia',
```

**Review by simenandre — CHANGES_REQUESTED** [review]
In general, `name` should be pascal case (e.g. `pascalCase`, `helloWorld`, `thisIsAString`)

**Review by simenandre — COMMENTED** [review]
Btw, let's go with _metadata_ not _meta data_

**Review by braaar — COMMENTED** [review]
Are we going to set the website `title` here in sanity or in the source code of the website?

**Pull request closed without merging**